lib/BackgroundJob/ExpirePageTrash.php

```diff
@@ -14,18 +14,20 @@
 class ExpirePageTrash extends TimedJob {
 	private Expiration $expiration;
 	private PageTrashBackend $trashBackend;
+	private bool $hasTrashBackend = false;
 
 	public function __construct(
-		ITimeFactory $time,
-		Expiration $expiration,
-		PageTrashBackend $trashBackend) {
+		ITimeFactory $time) {
```

Comment:
Starting with nextcloud 28 you could actually make them nullable, but good as it is as we cover more versions ;)

https://github.com/nextcloud/server/pull/41265

Reply:
Uh, that's great to learn. Not for now, but one day we'll drop support for everything below 28 :blush: